### PR TITLE
Generate loan summary on home page

### PR DESCRIPTION
**Pull request by adamico**
**Summary of changes**
1.  Realtime calculation of loan summary details when all fields are filled in (date validation check on loan start date field)
2.  Prepared to integrate loan details -- monthly/yearly amortization table -- will be added later maybe through a modal (showing this by default clutters the homepage)
3.  Added a spec with javascript enabled to test loan summary calculations (I added the `poltergeist` gem and loaded this in Rspec as the javascript driver)

### CODE REVIEW

**Comment by akshatpradhan** [comment]
Given I'm on '/'
Then I should see the Loan Summary box to the right of the arrow, but I don't. 
screen shot 2013-11-14 at 3 46 45 pm

Is it possible to make that happen?

**Comment by adamico** [comment]
The summary box is hidden until you fill in all the input fields

**Comment by akshatpradhan** [comment]
@adamico Let me rephrase the question, is it possible to do this: Imagine the box showed up on the right, but has nil values. Only when you start filling out the Loan Amount and finish with the First Payment Date, then the values populate, but the during that process, the Loan Summary box remains visible.

**Comment by akshatpradhan** [comment]
@adamico Also can you update your master branch and then do a `git checkout loan_summary && git rebase master`. There will probably be conflicts in notes/show.html, and then I can merge it in after a `git push --force origin loan_summary`

**Comment by adamico** [comment]
Sure I'll do that as soon as I can.

Andrea D'Amico

**Comment by adamico** [comment]
Here you are @akshatpradhan. Not very pretty when loan summary table is empty. I've removed the hide/show javascript animations until we figure out something prettier.
@alixaxel what do you mean by protocol-relative? The links are working fine over http, do you think rails server would complain for https pages? Is that what you mean?

**Comment by akshatpradhan** [comment]
@adamico You're right, its not very pretty when loan summary table is empty. I need to think harder about this design. Let's leave this for now until a better IxD can be figured out. Also, I noticed that you put the entire JavaScript in the index.html.erb page. Shouldn't we NOT put JS in the index.html.erb but put it into its own separate file (e.g. home.js.coffee?). That also makes it easier to test. I was curious what your opinion about that was.

**Comment by alixaxel** [comment]
@adamico The browser will complain if you're loading HTTP links over an HTTPS connection.

Just do it like this:

``` erb
<%= javascript_include_tag "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.4.0/moment.min.js" %>
<%= javascript_include_tag "//cdnjs.cloudflare.com/ajax/libs/accounting.js/0.3.2/accounting.min.js" %>
<%= javascript_include_tag "//cdnjs.cloudflare.com/ajax/libs/handlebars.js/1.1.2/handlebars.min.js" %>
```

**Comment by adamico** [comment]
@alixaxel yeah, that's what I thought you meant. I'll correct those.
@akshatpradhan I'll move the inline javascript to a separate coffeescript asset

**Comment by akshatpradhan** [comment]
## Change Description - :smile:

### What does this change do?

This features allows people to calculate the loan summary right on the front page. This is a client side feature that doesn't touch the database.

### What testing has been done?

This has been QA'd locally and appropriate tests have been run.

## Risk Assessment

### What are the risks? Is there a business impact?

There is no security risk since this is a client side loan calculator.
